### ISSUE TITLE
How can you measure the mean and standard deviation value ?

### ISSUE THREAD

**Issue by SamMohel** [zarzouram/image_captioning_with_transformers#6]
Thanks for sharing the code , excuse me if I need to measure the the mean and standard deviation value for the model with different initialization of the weights , Does that mean I need to train the model with different learning rates ? or I need to train the model three times for example as each time I train the model i got different accuracy but is a little different and evaluate the model after each time and get BLEU scores then calculate the average of the scores ?

**Comment by zarzouram** [comment]
Hello, Thank you for your question. Sorry for the late reply. 

I am still learning, so I am not an expert. But as far as I know, you start with tuning your hyperparameter (learning rate, batch size, and so on), with fixed initialization over your runs. Once you know your best hyperparameter (the way of weight initialization could be one of them), you can start changing the initialization weights and data split (by changing the seed) to calculate the mean and standard deviation. 

You evaluate/test the model after each training run and record the corresponding metric values. Then, you can calculate the mean and standard deviation of these values over all the runs.

Please note that this code is for a project for a course I took. I did not tune the hyperparameters or estimate the model's average performance.
